modules/tinymce/src/core/main/ts/lookup/UserLookup.ts

```diff
@@ -60,14 +62,34 @@ const userSchema = StructureSchema.objOf([
   FieldSchema.option('custom')
 ]);
 
+const objectCat = <T extends Record<string, any>>(
+  obj: { [K in keyof T]: Optional<T[K]> }
+): Partial<T> => {
+  const result = {} as Partial<T>;
+
+  Obj.each(obj, (value, key) => {
+    value.each((v) => {
+      result[key as keyof T] = v;
+    });
+  });
+
+  return result;
+};
+
+const extractOptionalField = <T>(field: any): Optional<T> =>
+  field?.fold(Optional.none, Optional.some) ?? Optional.none();
+
 const transformResult = (user: any): User => {
-  const customValue = user.custom?.getOrUndefined();
+  const optionalFields: { [K in keyof OptionalUserFields]: Optional<OptionalUserFields[K]> } = {
+    name: extractOptionalField(user.name),
+    avatar: extractOptionalField(user.avatar),
+    description: extractOptionalField(user.description),
+    custom: extractOptionalField(user.custom)
+  };
+
   return {
     id: user.id,
-    name: user.name?.getOrUndefined(),
-    avatar: user.avatar?.getOrUndefined(),
-    description: user.description?.getOrUndefined(),
-    ...(customValue !== undefined && { custom: customValue })
+    ...objectCat(optionalFields)
   };
```

Comment:
I think you got yourself a bit confused here. `extractOptionalField` is just `Optional.from()`.

But the whole point of my suggestion to create `objectCat` was to replace all this. Literally the entire `transformResult` function could just be a single call to `objectCat`. `transformResult` itself may not even be needed, just call `objectCat` at the call site instead.